network-config-analyzer/ResourcesHandler.py

```diff
@@ -0,0 +1,256 @@
+
+        # parse for policies
+        resources_parser.parse_lists_for_policies(np_list, resource_list, peer_container, k8s_np_flag)
+
+        if config_name == 'global':
+            if np_list and np_list != ['']:
+                config_name = np_list[0]
+            elif resource_list:
+                config_name = resource_list[0]
+        # build and return the networkConfig
+        return NetworkConfig(name=config_name, peer_container=peer_container,
+                             policies_container=resources_parser.policies_finder.policies_container,
+                             config_type=resources_parser.policies_finder.type)
+
+    def set_global_peer_container(self, global_ns_list, global_pod_list, global_resource_list):
+        """
+        builds the global peer container based on global input resources,
+        it also saves the global pods and namespaces finder, to use in case specific configs missing one of them
+        :param Union[list[str], None] global_ns_list: global namespaces entries
+        :param Union[list[str], None] global_pod_list: global pods entries
+        :param Union[list[str], None] global_resource_list: list of global entries of namespaces/pods to handle
+        in case specific list is None
+        """
+        resources_parser = ResourcesParser()
+        success, res_type = resources_parser.parse_lists_for_topology(global_ns_list, global_pod_list,
+                                                                      global_resource_list)
+        if success:
+            self.global_peer_container = resources_parser.build_peer_container()
+            self.global_pods_finder = resources_parser.pods_finder
+            self.global_ns_finder = resources_parser.ns_finder
+        elif res_type == ResourceType.Pods:
+            # in case the global resources has only pods (can not build global peerContainer)
+            self.global_pods_finder = resources_parser.pods_finder
+        elif res_type == ResourceType.Namespaces:
+            # in case the global resources has only namespaces (can not build global peerContainer)
+            self.global_ns_finder = resources_parser.ns_finder
+
+
+class ResourcesParser:
+    """
+    This class parses the input resources for topology (pods, namespaces, services) and policies.
+    """
+    def __init__(self):
+        self.policies_finder = PoliciesFinder()
+        self.pods_finder = PodsFinder()
+        self.ns_finder = NamespacesFinder()
+        self.services_finder = ServicesFinder()
+
+    @staticmethod
+    def _determine_topology_resource(topology_list, resource_list, res_type):
+        res_name = 'namespaces' if res_type == ResourceType.Namespaces else 'pods and services'
+        if topology_list:
+            topology_resource = topology_list
+            if resource_list:
+                print(f'Warning: {res_name} will be taken from the provided namespace list key.'
```

Comment:
The content of the warning says "will be taken from the provided **namespace**  list key" , but it's not always namespace